src/backend/distributed/metadata/node_metadata.c

```diff
@@ -759,6 +759,15 @@ master_update_node(PG_FUNCTION_ARGS)
 			handle = StartLockAcquireHelperBackgroundWorker(MyProcPid, lock_cooldown);
 		}
 
+		if (!handle)
+		{
+			StringInfo lock_cooldown_string_value = makeStringInfo();
+			appendStringInfo(lock_cooldown_string_value, "\"%d\"", lock_cooldown);
+			set_config_option("lock_timeout", lock_cooldown_string_value->data,
+							  (superuser() ? PGC_SUSET : PGC_USERSET), PGC_S_SESSION,
+							  GUC_ACTION_LOCAL, true, 0, false);
```

Comment:
this can go into a function like we do for other similar functions like `SetLocalShardModifyMode`